test/DomainRegister.test.ts

```diff
@@ -0,0 +1,72 @@
+import { expect } from "chai";
+import { ethers } from "hardhat";
+import { Contract } from "ethers";
+
+describe("DomainRegister", function () {
+  let DomainRegister: Contract;
+  let domainRegister: Contract;
+  let owner: Contract;
+  let addr1: Contract;
+  let addr2: Contract;
+
+  beforeEach(async function () {
+    DomainRegister = await ethers.getContractFactory("DomainRegister");
+    [owner, addr1, addr2] = await ethers.getSigners();
+    domainRegister = await DomainRegister.deploy(ethers.parseEther("1"));
+    await domainRegister.waitForDeployment();
```

Comment:
It may be not mandatory to wait for contract receipt for tests, when Hardhat runs transactions on demand